.github/workflows/check-and-gen.yml

```diff
@@ -70,3 +68,38 @@ jobs:
       - uses: ./.github/actions/install-go
       - run: go mod tidy -v
       - run: git diff --exit-code
+  generate-docs:
+    name: Generate Documentation
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/release/'))
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+          fetch-depth: 0
+
+      - uses: ./.github/actions/install-system-dependencies
+      - uses: ./.github/actions/install-go
+      - uses: ./.github/actions/make-deps
+
+      - name: Generate API documentation using docsgen-cli
+        run: |
+          make docsgen-cli || {
+            echo "Error: Documentation generation failed"
+            exit 1
+          }
+
+      - name: Commit and push if documentation changed
+        run: |
+          if [ -n "$(git status --porcelain)" ]; then
+           # Bot email is configured from: https://github.com/orgs/community/discussions/26560
+            git config user.name "github-actions[bot]"
+            git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+            git add .
+            git commit -m "docs: update API documentation via docsgen-cli"
+            git push
```

Comment:
Another thing I'm wondering is if this is going to work to fork-branches, if this is run a a filecoin-project GHA runner, will we always have permissions to push back into the PR branch? Do we need to ensure we have a "can edit branches" thing turned on in our repo? Do we need to do anything else to make this work?

Reply:
As currently configured, this will be run on push events to `master` or `release/*`. But it won't work because they are protected. To overcome this, one would have to use a personal access token for a user that can disregard the branch protection rules.